app/src/main/java/com/github/se/eventradar/viewmodel/EventDetailsViewModel.kt

```diff
@@ -99,159 +103,77 @@ constructor(
       }
 
       if (!_isUserAttending.value) {
-        // TODO needs to be atomic to avoid concurrency issues
-        registrationUpdateEvent()
-        if (!errorOccurred.value) {
-          registrationUpdateUser()
+        if (registrationProcedure(eventId, currentUserId)) {
+          registrationSuccessful.value = true
```

Comment:
question: is this still needed? it seems in the file, it's only called here

Reply:
yes, it is used by the UI in EventDetails to display the Success Dialog box